domain/transaction.go

```diff
@@ -65,31 +85,37 @@ func (TransactionHandler) RouteCommandToInstance(m dogma.Message) string {
 
 // HandleCommand handles a command message that has been routed to this
 // handler.
-func (TransactionHandler) HandleCommand(s dogma.AggregateCommandScope, m dogma.Message) {
+func (TransactionHandler) HandleCommand(
+	r dogma.AggregateRoot,
+	s dogma.AggregateCommandScope,
+	m dogma.Message,
+) {
+	t := r.(*transaction)
+
 	switch x := m.(type) {
 	case commands.Deposit:
-		startDeposit(s, x)
+		startDeposit(t, s, x)
 	case commands.ApproveDeposit:
-		approveDeposit(s, x)
+		approveDeposit(t, s, x)
 	case commands.Withdraw:
-		startWithdraw(s, x)
+		startWithdraw(t, s, x)
 	case commands.ApproveWithdrawal:
-		approveWithdrawal(s, x)
+		approveWithdrawal(t, s, x)
 	case commands.DeclineWithdrawal:
-		declineWithdrawal(s, x)
+		declineWithdrawal(t, s, x)
 	case commands.Transfer:
-		startTransfer(s, x)
+		startTransfer(t, s, x)
 	case commands.ApproveTransfer:
-		approveTransfer(s, x)
+		approveTransfer(t, s, x)
 	case commands.DeclineTransfer:
-		declineTransfer(s, x)
+		declineTransfer(t, s, x)
 	default:
 		panic(dogma.UnexpectedMessage)
 	}
 }
 
-func startDeposit(s dogma.AggregateCommandScope, m commands.Deposit) {
-	if !s.Create() {
+func startDeposit(t *transaction, s dogma.AggregateCommandScope, m commands.Deposit) {
+	if t.Started {
 		s.Log("transaction already exists")
```

Comment:
```suggestion
		s.Log("transaction already started")
```

Since we're changing the idempotency checks to use business concepts, probs worth changing the language in the log as well.